### PR TITLE
Hybrid Cache: Element cache

### PR DESCRIPTION
**Pull request by lauraneto**
## Summary

Fully implements the element hybrid cache — the elements equivalent of the existing document and media hybrid caches. Elements now get the same multi-level caching (L0 converted element cache, L1/L2 HybridCache, database cache table) with notification-based invalidation.

### Element cache service
- `ElementCacheService` with full HybridCache backing + `cmsContentNu` database persistence
- Draft/published separation, preview service support, cache seeding infrastructure
- `IContentCacheService` base interface extracted and shared across document, media, and element cache services
- `IPublishedElementCache` public facade with async-only API, exposed via `ICacheManager.Elements` and `IUmbracoContext.Elements`

### Database layer
- Element CRUD methods on `IDatabaseCacheRepository` (`GetElementSource*`, `RefreshElementAsync`, `RebuildElementDbCache`)
- `SqlElementSourcesSelect` query using `ElementDto`/`ElementVersionDto`
- `ContentCacheDataSerializerEntityType.Element` for serialization
- Renamed document-specific methods for clarity (`GetContentSource` → `GetDocumentSource`, etc.)
- Extracted shared `RebuildPublishableDbCache` to eliminate duplication between document and element rebuild

### Cache invalidation
- `ElementRefreshNotification` fired from `ElementRepository.OnUowRefreshedEntity`
- `CacheRefreshingNotificationHandler` handles element notifications, routes content type changes by `IsElement`
- Refactored handler to single-pass classification with `RefreshCacheForContentTypeChanges`

### Navigation and seeding
- `ElementNavigationService` with `IElementNavigationQueryService`/`IElementNavigationManagementService`
- Queries both `Element` and `ElementContainer` object types to build full tree with containers
- Multi-objectType overloads on `INavigationRepository` with LEFT JOIN for container support
- `ElementBreadthFirstKeyProvider` traverses containers without seeding them, only counting published elements
- `ElementContentTypeSeedKeyProvider` seeds elements by configured content type keys
- Configurable `ElementBreadthFirstSeedCount` and `ElementSeedBatchSize` in `CacheSettings`

### Publish status
- Split `PublishStatusService` into abstract base + `DocumentPublishStatusService` / `ElementPublishStatusService`
- Separate interfaces: `IDocumentPublishStatusQueryService`, `IElementPublishStatusQueryService`, etc.
- Element publish status initialized on startup and kept in sync via `ElementCacheRefresher`
- Non-breaking constructor changes with obsolete overloads for backward compatibility

### Bug fix
- Fixed `CacheNodeFactory.ToContentCacheNode(IElement)` hardcoding `IsDraft = false` instead of using the `preview` parameter — this prevented draft `cmsContentNu` rows from being written

### Tests
- `CacheRefreshingNotificationHandlerTests` — 8 tests including element type routing
- `ElementCacheServiceTests` — 9 integration tests (rebuild, delete, refresh)
- `ElementHybridCacheTests` — 7 integration tests via `IPublishedElementCache`
- `ElementHybridCacheElementTypeTests` — 3 integration tests for type change invalidation
- `ElementBreadthFirstKeyProviderTests` — 9 unit tests including container traversal

## Breaking changes

- **`PublishStatusService`** is now an abstract base class. Use `DocumentPublishStatusService` or `ElementPublishStatusService` instead of instantiating it directly.
- **`IPublishStatusQueryService`** is now obsolete — use `IDocumentPublishStatusQueryService` or `IElementPublishStatusQueryService` directly.
- **`IPublishStatusManagementService`** is now obsolete — use `IDocumentPublishStatusManagementService` or `IElementPublishStatusManagementService` directly.
- **`ICacheManager`** has a new `Elements` property (`IPublishedElementCache`). Implementors must add this property.
- **`IUmbracoContext`** has a new `Elements` property (`IPublishedElementCache`). Implementors must add this property.
- **`CacheManager` constructor** now requires an `IPublishedElementCache` parameter.
- **`INavigationRepository`** has new multi-objectType overloads: `GetContentNodesByObjectType(IEnumerable<Guid>)` and `GetTrashedContentNodesByObjectType(IEnumerable<Guid>)`. The single-type overloads now delegate to these.
- **`IDatabaseCacheRepository.Rebuild`** signature changed — added optional `elementTypeIds` parameter.
- **`IDocumentCacheService`**, **`IMediaCacheService`**, **`IElementCacheService`** now extend `IContentCacheService` (additive, but implementors of these interfaces now implicitly implement `IContentCacheService`).
- **`IElementCacheService`** interface expanded with new methods: `RefreshElementAsync`, `DeleteItemAsync`, `Rebuild`, `RebuildMemoryCacheByContentTypeAsync`, `ClearConvertedContentCache`, `GetByContentType`, `SeedAsync`.
- **Renamed repository methods**: `GetContentSourceAsync` → `GetDocumentSourceAsync`, `RefreshContentAsync` → `RefreshDocumentAsync`, etc. (internal interface, but affects custom `IDatabaseCacheRepository` implementations).
- **`ContentNavigationRepository`** single-type query now uses LEFT JOIN instead of INNER JOIN (behavioral change — may return nodes without `umbracoContent` rows).

## Known issues

> [!WARNING]
> Trashing an item does not clear it from the published cache. A cached element (or document) remains accessible via the cache after being moved to the recycle bin. This is a pre-existing issue that also affects documents — it is not introduced by this PR. The `Cannot_Get_Published_Again_After_Trashing` test has been skipped pending a fix.

### CODE REVIEW

File: src/Umbraco.Core/Models/CacheSettings.cs [hunk 1]
```diff
@@ -52,6 +61,12 @@ public class CacheSettings
     [DefaultValue(StaticMediaBreadthFirstSeedCount)]
     public int MediaBreadthFirstSeedCount { get; set; } = StaticDocumentBreadthFirstSeedCount;
```
[the remark this answers is not in the document]
Comment (lauraneto) [reply src/Umbraco.Core/Models/CacheSettings.cs]: Existing issue. Will make a separate PR directly to main.

File: src/Umbraco.Core/Cache/Refreshers/Implement/ContentCacheRefresher.cs [hunk 2]
```diff
@@ -36,13 +36,51 @@ public sealed class ContentCacheRefresher : PayloadCacheRefresherBase<ContentCac
...
+        // TODO: Ideally we should inject IElementsCache
+        // this interface is in infrastructure, and changing this is very breaking
+        // so as long as we have the cache manager, which casts the IElementsCache to a simple AppCache we might as well use that.
+        _cacheManager = cacheManager;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ContentCacheRefresher"/> class.
+    /// </summary>
+    [Obsolete("Please use the non-obsolete constructor. Scheduled for removal in Umbraco 19.")]
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Cache/Refreshers/Implement/ContentCacheRefresher.cs:83]: For a constructor on a class that's not expected to be instantiated in project code, this is OK.

File: src/Umbraco.Core/Cache/Refreshers/Implement/ElementCacheRefresher.cs [hunk 3]
```diff
@@ -169,6 +171,27 @@ private void HandleMemoryCache(JsonPayload payload)
         }
     }
 
+    private void HandlePublishStatus(JsonPayload payload)
```
Comment (AndyButland) [inline src/Umbraco.Core/Cache/Refreshers/Implement/ElementCacheRefresher.cs]: In `ContentCacheRefresher` this is named `HandlePublishedAsync`.  If you prefer the naming and return type here - I think I do - you can update `ContentCacheRefresher` to align.,

File: src/Umbraco.Core/Models/CacheSettings.cs [hunk 4]
```diff
@@ -52,6 +61,12 @@ public class CacheSettings
...
 
+    /// <summary>
+    /// Gets or sets a value for the element breadth first seed count.
+    /// </summary>
+    [DefaultValue(StaticElementBreadthFirstSeedCount)]
+    public int ElementBreadthFirstSeedCount { get; set; } = StaticElementBreadthFirstSeedCount;
```
Comment (AndyButland) [inline src/Umbraco.Core/Models/CacheSettings.cs:68]: Hopefully we are already tracking somewhere docs updates that are needed for global elements.  Please make sure these configuration updates are included.

File: src/Umbraco.Core/Notifications/ElementRefreshNotification.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+/// <summary>
+///     Notification published when an element needs to be refreshed in the cache.
+/// </summary>
+/// <remarks>
+///     This notification is used internally for cache synchronization purposes.
+///     For normal element operations, use <see cref="ElementSavedNotification"/> or
+///     <see cref="ElementPublishedNotification"/> instead.
+/// </remarks>
+[Obsolete("This is only used for the internal cache and will change, use saved notifications instead")]
```
Comment (AndyButland) [inline src/Umbraco.Core/Notifications/ElementRefreshNotification.cs:18]: Is there a plan to remove this?  If so, should the obsolete message have a version? If not, should we remove the `Obsolete`?

File: src/Umbraco.Core/Persistence/Repositories/INavigationRepository.cs [hunk 6]
```diff
@@ -14,10 +14,24 @@ public interface INavigationRepository
     /// <returns>A collection of navigation models.</returns>
     IEnumerable<INavigationModel> GetContentNodesByObjectType(Guid objectTypeKey);
 
+    /// <summary>
+    ///     Retrieves a collection of nodes as navigation models for multiple object types.
+    /// </summary>
+    /// <param name="objectTypeKeys">The unique identifiers for the object types to include.</param>
+    /// <returns>A collection of navigation models.</returns>
+    IEnumerable<INavigationModel> GetContentNodesByObjectType(IEnumerable<Guid> objectTypeKeys);
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Persistence/Repositories/INavigationRepository.cs:22]: This is OK for an update due in a major.

File: src/Umbraco.Core/Persistence/Repositories/INavigationRepository.cs [hunk 7]
```diff
@@ -14,10 +14,24 @@ public interface INavigationRepository
...
     /// <param name="objectTypeKey">The unique identifier for the object type.</param>
     /// <returns>A collection of navigation models.</returns>
     IEnumerable<INavigationModel> GetTrashedContentNodesByObjectType(Guid objectTypeKey);
+
+    /// <summary>
+    ///     Retrieves a collection of trashed nodes as navigation models for multiple object types.
+    /// </summary>
+    /// <param name="objectTypeKeys">The unique identifiers for the object types to include.</param>
+    /// <returns>A collection of navigation models.</returns>
+    IEnumerable<INavigationModel> GetTrashedContentNodesByObjectType(IEnumerable<Guid> objectTypeKeys);
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Persistence/Repositories/INavigationRepository.cs:36]: This is OK for an update due in a major.

File: src/Umbraco.Core/Persistence/Repositories/IPublishStatusRepository.cs [hunk 8]
```diff
@@ -27,4 +27,19 @@ public interface IPublishStatusRepository
     /// <param name="cancellationToken">A cancellation token to observe.</param>
     /// <returns>A dictionary mapping document keys to their published culture codes.</returns>
     Task<IDictionary<Guid, ISet<string>>> GetDescendantsOrSelfPublishStatusAsync(Guid rootDocumentKey, CancellationToken cancellationToken);
+
+    /// <summary>
+    ///     Gets the publish status for all elements.
+    /// </summary>
+    /// <param name="cancellationToken">A cancellation token to observe.</param>
+    /// <returns>A dictionary mapping element keys to their published culture codes.</returns>
+    Task<IDictionary<Guid, ISet<string>>> GetAllElementPublishStatusAsync(CancellationToken cancellationToken);
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Persistence/Repositories/IPublishStatusRepository.cs:36]: This is OK for an update due in a major.

File: src/Umbraco.Core/PublishedCache/ICacheManager.cs [hunk 9]
```diff
@@ -27,6 +27,11 @@ public interface ICacheManager
     /// </summary>
     IPublishedMemberCache Members { get; }
 
+    /// <summary>
+    ///     Gets the <see cref="IPublishedElementCache" />.
+    /// </summary>
+    IPublishedElementCache Elements { get; }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/PublishedCache/ICacheManager.cs:33]: This is OK for an update due in a major.

File: src/Umbraco.Core/Web/IUmbracoContext.cs [hunk 10]
```diff
@@ -41,6 +41,11 @@ public interface IUmbracoContext : IDisposable
     /// </summary>
     IPublishedMediaCache Media { get; }
 
+    /// <summary>
+    ///     Gets the published element cache.
+    /// </summary>
+    IPublishedElementCache Elements { get; }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Web/IUmbracoContext.cs:47]: This is OK for a PR due for a major release.

File: src/Umbraco.PublishedCache.HybridCache/Persistence/IDatabaseCacheRepository.cs [hunk 11]
```diff
@@ -8,46 +8,33 @@ namespace Umbraco.Cms.Infrastructure.HybridCache.Persistence;
 /// </summary>
 internal interface IDatabaseCacheRepository
 {
-    /// <summary>
-    /// Deletes the specified content item from the cache database.
-    /// </summary>
-    Task DeleteContentItemAsync(int id);
+    // --- Document methods ---
```
Comment (AndyButland) [inline src/Umbraco.PublishedCache.HybridCache/Persistence/IDatabaseCacheRepository.cs]: I'm not a huge fan of them, but in this case maybe regions would add structure to this class better than these inline comments?

File: tests/Umbraco.Tests.Integration/Umbraco.Core/Services/ElementPublishingServiceTests.Publish.cs [hunk 12]
```diff
@@ -140,6 +140,7 @@ public async Task Cannot_Get_Trashed_As_Published()
     }
 
     [Test]
+    [Ignore("Trashing does not clear the published cache - this is a pre-existing issue that also affects documents.")]
```
Comment (AndyButland) [inline tests/Umbraco.Tests.Integration/Umbraco.Core/Services/ElementPublishingServiceTests.Publish.cs]: Just checking you have an action to restore this. This is what you fixed in 17 right? So perhaps you can now merge up and enable this test.

File: src/Umbraco.PublishedCache.HybridCache/Services/ElementCacheService.cs [hunk 13]
```diff
@@ -1,58 +1,347 @@
...
+                continue;
+            }
+
+            using ICoreScope scope = _scopeProvider.CreateCoreScope();
+
+            IEnumerable<ContentCacheNode> cacheNodes = await _databaseCacheRepository.GetElementSourcesAsync(uncachedKeys);
+
+            scope.Complete();
+
+            _logger.LogDebug("Element nodes to cache {NodeCount}", cacheNodes.Count());
```
Comment (AndyButland) [inline src/Umbraco.PublishedCache.HybridCache/Services/ElementCacheService.cs]: There's a double enumeration here and in the `foreach` below.  Suggest something like `var cacheNodesList = cacheNodes.ToList();` before the log line so we only iterate once.  The issue has been duplicated from `DocumentCacheService`, so can be fixed there too.

**Review by AndyButland — CHANGES_REQUESTED** [review]
Looks very good to me @lauraneto.  I spotted a few things inline for check-over.

Test coverage looks great, but I noticed for documents we have `DocumentHybridCacheVariantsTests`, `DocumentHybridCacheScopeTests`, `DocumentPropertyCacheLevelTests` and `DocumentHybridCacheMockTests` and no equivalent for elements.  Maybe that's worth looking over and seeing if it makes sense to create element versions.  No need to do so if you don't see any value though.  The main gap looks like tests for culture variants.

I've limited myself to code review here and not done any manual testing - I think that given automated testing is solid, any further edge cases will come out in the wash as the feature as a whole is developed and tested in advance of 18 release.

There was previously a build failure but the error message referred to a file that didn't exist, so I wasn't sure what was going on there.  Have pushed an empty commit to trigger the build again.

File: src/Umbraco.Core/Persistence/Repositories/IPublishStatusRepository.cs [hunk 14]
```diff
@@ -27,4 +27,19 @@ public interface IPublishStatusRepository
...
+
+    /// <summary>
+    ///     Gets the publish status for a specific element.
+    /// </summary>
+    /// <param name="elementKey">The unique key of the element.</param>
+    /// <param name="cancellationToken">A cancellation token to observe.</param>
+    /// <returns>A set of culture codes for which the element is published.</returns>
+    Task<ISet<string>> GetElementPublishStatusAsync(Guid elementKey, CancellationToken cancellationToken);
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Persistence/Repositories/IPublishStatusRepository.cs:44]: This is OK in a PR targeted for a major.

File: src/Umbraco.Core/Notifications/ElementRefreshNotification.cs (hunk 5 above)
Comment (lauraneto) [reply src/Umbraco.Core/Notifications/ElementRefreshNotification.cs:18]: Doesn't look like it. It was copied from the other refresh notifications and it seems to be there since 2021.
I think it was just a "warning" that this is public but that doesn't mean consumers should use it. Not sure if there is a better way to annotate that.

File: tests/Umbraco.Tests.Integration/Umbraco.Core/Services/ElementPublishingServiceTests.Publish.cs (hunk 12 above)
Comment (lauraneto) [reply tests/Umbraco.Tests.Integration/Umbraco.Core/Services/ElementPublishingServiceTests.Publish.cs]: Oh, yes, will look into it!

File: src/Umbraco.Core/Notifications/ElementRefreshNotification.cs (hunk 5 above)
Comment (AndyButland) [reply src/Umbraco.Core/Notifications/ElementRefreshNotification.cs:18]: Then let's leave it then.  I hadn't realised it was something already existing in related areas.

File: src/Umbraco.PublishedCache.HybridCache/Persistence/IDatabaseCacheRepository.cs (hunk 11 above)
Comment (lauraneto) [reply src/Umbraco.PublishedCache.HybridCache/Persistence/IDatabaseCacheRepository.cs]: Adjusted.

File: src/Umbraco.PublishedCache.HybridCache/Services/ElementCacheService.cs (hunk 13 above)
Comment (lauraneto) [reply src/Umbraco.PublishedCache.HybridCache/Services/ElementCacheService.cs]: Adjusted.

File: src/Umbraco.Core/Cache/Refreshers/Implement/ElementCacheRefresher.cs (hunk 3 above)
Comment (lauraneto) [reply src/Umbraco.Core/Cache/Refreshers/Implement/ElementCacheRefresher.cs]: I adjusted `ContentCacheRefresher` to `HandlePublishStatusAsync`, but also adjusted this one to be async to match.

File: src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentNavigationRepository.cs [hunk 15]
```diff
@@ -51,9 +66,10 @@ private IEnumerable<INavigationModel> FetchNavigationDtos(Guid objectTypeKey, bo
                 $"n.{syntax.GetQuotedColumnName(NodeDto.SortOrderColumnName)}  as  {syntax.GetQuotedColumnName(NodeDto.SortOrderColumnName)}",
                 $"n.{syntax.GetQuotedColumnName(NodeDto.TrashedColumnName)}  as  {syntax.GetQuotedColumnName(NodeDto.TrashedColumnName)}")
             .From<NodeDto>("n")
-            .InnerJoin<ContentDto>("c").On<NodeDto, ContentDto>((n, c) => n.NodeId == c.NodeId, "n", "c")
-            .InnerJoin<NodeDto>("ctn").On<ContentDto, NodeDto>((c, ctn) => c.ContentTypeId == ctn.NodeId, "c", "ctn")
-            .Where<NodeDto>(n => n.NodeObjectType == objectTypeKey && n.Trashed == trashed, "n")
+            .LeftJoin<ContentDto>("c").On<NodeDto, ContentDto>((n, c) => n.NodeId == c.NodeId, "n", "c")
+            .LeftJoin<NodeDto>("ctn").On<ContentDto, NodeDto>((c, ctn) => c.ContentTypeId == ctn.NodeId, "c", "ctn")
```
Comment (AndyButland) [inline src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentNavigationRepository.cs]: Do you think we should keep the inner join behaviour for object types other than elements? I can see why we need this for elements due to the element containers, but now we do this looser join for document and media too, which should always have a row in `umbracoContent`.  There could be some SQL query plan optimisation we lose here too by effectively giving less information to the optimizer by using OUTER over INNER joins.

File: src/Umbraco.Core/Services/PublishStatus/DocumentPublishStatusService.cs [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+        PopulateCache(publishStatus);
+        DefaultCulture = await _languageService.GetDefaultIsoCodeAsync();
+    }
+
+    /// <inheritdoc/>
+    public bool IsDocumentPublished(Guid documentKey, string culture)
+        => IsPublished(documentKey, culture);
+
+    /// <inheritdoc/>
+    public bool IsDocumentPublishedInAnyCulture(Guid documentKey)
```
Comment (AndyButland) [inline src/Umbraco.Core/Services/PublishStatus/DocumentPublishStatusService.cs]: These names differ from the equivalents in `ElementPublishStatusService`, where we have `IsPublished` and `IsPublishedInAnyCulture`.  Worth aligning them?  I would say the `Document` part of the method names here is unnecessary as it's clear from the class that we are working with documents, so these would be the ones to change.  If you need to, you can obsolete these methods and delegate them to the aligned named equivalents you introduce.

File: src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+/// </summary>
+public interface IPublishedElementCache
+{
+    /// <summary>
+    /// Gets a published element by its unique key.
+    /// </summary>
+    /// <param name="key">The element's unique key.</param>
+    /// <param name="preview">Optional value indicating whether to include unpublished content.</param>
+    /// <returns>The published element, or <c>null</c> if not found.</returns>
+    Task<IPublishedElement?> GetByIdAsync(Guid key, bool? preview = null);
```
Comment (AndyButland) [inline src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs:16]: Should this be `GetByKeyAsync`? I know we are moving away from integers to GUIDs, but we still have both, so it might be clearer to keep alignment with other services that do offer gets by Id (integer) and key (GUID)?

File: src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+        IOptions<CacheSettings> cacheSettings,
+        IElementPublishStatusQueryService publishStatusService)
+        : base(elementNavigationQueryService, cacheSettings.Value.ElementBreadthFirstSeedCount)
+    {
+        _publishStatusService = publishStatusService;
+        _seedCount = cacheSettings.Value.ElementBreadthFirstSeedCount;
+    }
+
+    /// <inheritdoc/>
+    public new ISet<Guid> GetSeedKeys()
```
Comment (AndyButland) [inline src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs]: Could be there is some clean-up that could be done here.

- We are using new, as the base class isn't virtual.  But we could make it so, and override instead.
- But then I see a TODO for 16 on this same method in `DocumentBreadthFirstKeyProvider` to move this to the base class. Clearly that wasn't done.  It does seem that particularly with the introduction for elements, there's a fair bit of code duplicated.  So maybe some or all of it could be moved to the base class now, and we have re-use across the documents and elements.

**Review by AndyButland — CHANGES_REQUESTED** [review]
Thanks for the updates @lauraneto - it's all looking good, though I found a few other things to consider on a second pass. Please see what you think.

File: src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs (hunk 17 above)
Comment (lauraneto) [reply src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs:16]: I've adjusted this for elements, although it is now not in line with documents (`IPublishedContentCache`) and media (`IPublishedMediaCache`), which use `id` for both int and key. Should we also align those?

File: src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs (hunk 18 above)
Comment (lauraneto) [reply src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs]: I've adjusted it here: https://github.com/umbraco/Umbraco-CMS/pull/22369/changes/4b41b3ae74b818b308d51bae9ded10ea725999b9
Let me know your thoughts!

File: src/Umbraco.Core/Services/PublishStatus/DocumentPublishStatusService.cs (hunk 16 above)
Comment (lauraneto) [reply src/Umbraco.Core/Services/PublishStatus/DocumentPublishStatusService.cs]: Adjusted!

File: src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentNavigationRepository.cs (hunk 15 above)
Comment (lauraneto) [reply src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentNavigationRepository.cs]: I've done some changes in https://github.com/umbraco/Umbraco-CMS/pull/22369/commits/7a5484687bfdc302b55bbc767183911280d39fee, but not sure if it aligns with what you were thinking? Let me know!

Comment (AndyButland) [reply src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentNavigationRepository.cs]: Yes, that was what I had in mind - keeping the possibly more performant/data correct `INNER JOIN` for documents and media.

File: src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs (hunk 17 above)
Comment (AndyButland) [reply src/Umbraco.Core/PublishedCache/IPublishedElementCache.cs:16]: Hmm, so they do.  On reflection it's probably more important to align with those, and then changing them just means more churn going on.  So given we already have this slight mix of naming, maybe it was better how you had it originally.  Sorry!

File: src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs (hunk 18 above)
Comment (AndyButland) [reply src/Umbraco.PublishedCache.HybridCache/SeedKeyProviders/Element/ElementBreadthFirstKeyProvider.cs]: That looks great to me, much cleaner.

**Review by AndyButland — APPROVED** [review]
Great job @lauraneto - this all looks good to me and thanks for bearing with a couple of rounds of amends.  Just one conversation to resolve (which is going back on something I suggested before).
